Convert two external paintserver tests in svg/custom/ to reftests

Ahmad-S792 opened the pull request:
#### f8806ab8f6fe38ff0025e8ea27fceba22ff25715
<pre>
Convert two external paintserver tests in svg/custom/ to reftests
<a href="https://bugs.webkit.org/show_bug.cgi?id=284430">https://bugs.webkit.org/show_bug.cgi?id=284430</a>
<a href="https://rdar.apple.com/141255575">rdar://141255575</a>

Reviewed by Taher Ali.

Merge: <a href="https://source.chromium.org/chromium/chromium/src/+/6bfac6ae76afebd09ad6f141205b22643d7566a1">https://source.chromium.org/chromium/chromium/src/+/6bfac6ae76afebd09ad6f141205b22643d7566a1</a>

This patch is to make these two tests reftest.

Thanks Taher for helping in fixing `fuzziness` issue.

* LayoutTests/platform/gtk/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac-tahoe-wk2-lbse-text/svg/filters/feImage-remove-target-expected.txt: Removed.
* LayoutTests/platform/mac-tahoe-wk2-pixel/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/IndexedDB/idbcursor-advance-continue-async-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/external-paintserver-reference-expected.svg: Added.
* LayoutTests/svg/custom/external-paintserver-reference-expected.txt: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.svg: Added.
* LayoutTests/svg/custom/uri-reference-handling-expected.txt: Removed.
* LayoutTests/svg/custom/uri-reference-handling.svg:

Canonical link: <a href="https://commits.webkit.org/316417@main">https://commits.webkit.org/316417@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/907f47e49b087f5daca39fdd901698bfd599f6b7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/172242 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/46160 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/39588 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/181693 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/129798 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2b9bc0c5-eb05-4004-8f63-492a53f025b6) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/47453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/45802 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/133968 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/94511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/175200 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/37399 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/154516 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/114439 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/298ccc7a-044c-433d-ad24-3f0808ef53cb) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/36033 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/34301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/30299 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/146463 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/34027 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/184227 "Built successfully") | | 
| | | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/38503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/142732 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/45832 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/154099 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/142612 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/38504 "Built successfully and passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/46510 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/170/builds/30868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/111224 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/37685 "Passed tests") | | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/45027 "Built successfully") | [❌ 🧪 mac-site-isolation](https://ews-build.webkit.org/#/builders/185/builds/8967 "") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/44427 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/44659 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/44486 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 committed:
Convert two external paintserver tests in svg/custom/ to reftests

https://bugs.webkit.org/show_bug.cgi?id=284430
rdar://141255575

Reviewed by NOBODY (OOPS!).

Merge: https://source.chromium.org/chromium/chromium/src/+/6bfac6ae76afebd09ad6f141205b22643d7566a1

This patch is to make these two tests reftest.

* LayoutTests/platform/gtk/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/gtk/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/external-paintserver-reference-expected.txt: Removed.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/uri-reference-handling-expected.txt: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/external-paintserver-reference-expected.txt: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.txt: Removed.
* LayoutTests/svg/custom/external-paintserver-reference-expected.svg:
* LayoutTests/svg/custom/uri-reference-handling-expected.svg:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/49f1659510626e6f6b61b7ba5ff581e669343e00. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37753

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/314cfcb157fa4043b9f85380646a524b8f4a04b6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37753

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7e784b46cd017af2edd5cafc53a5595210f4dc1b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37753

Ahmad-S792 commented:
`[gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80630)` is flaky failure in View Transition test case - see it quite often on many PRs.

annevk commented on LayoutTests/svg/custom/uri-reference-handling.svg:
```diff
@@ -1,4 +1,5 @@
 <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" onload="runTest()">
+<meta name="fuzzy" content="maxDifference=1; totalPixels=0-2157" />
```
This hides regressions on certain platforms. E.g., the Skia rendering is currently as expected, but if it regresses we wouldn't catch that now.

Ahmad-S792 replied:
It is gradient noise - which WebKit adds (due to fingerprinting) and it leads to 1px differences and we fail some WPT due to it as well. See below, it is noise rather than regression and it is just 1px tolerance, so I don't agree that it is regression.

<img width="810" alt="image" src="https://github.com/user-attachments/assets/5e97833f-2832-4df8-9f7d-2217081ff28d" />

karlcow replied:
Yes there are plenty of WPT gradient tests failing because of this noise. 
Where fuzziness should be added.

annevk replied:
I thought we only added such noise to canvas?

Ahmad-S792 replied:
We definitely do it to gradients as well.

annevk commented on LayoutTests/svg/custom/external-paintserver-reference-expected.svg:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" standalone="no"?>
+<html xmlns="http://www.w3.org/1999/xhtml" style="height:100%"/>
```
I don't understand how this can be a valid reference given that `LayoutTests/svg/custom/external-paintserver-reference-expected.txt` lists a variety of colors that are no longer mentioned here.

Ahmad-S792 committed:
Convert two external paintserver tests in svg/custom/ to reftests
https://bugs.webkit.org/show_bug.cgi?id=284430
rdar://141255575

Reviewed by Taher Ali.

Merge: https://source.chromium.org/chromium/chromium/src/+/6bfac6ae76afebd09ad6f141205b22643d7566a1

This patch is to make these two tests reftest.

Thanks Taher for helping in fixing `fuzziness` issue.

* LayoutTests/platform/gtk/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac-tahoe-wk2-lbse-text/svg/filters/feImage-remove-target-expected.txt: Removed.
* LayoutTests/platform/mac-tahoe-wk2-pixel/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/IndexedDB/idbcursor-advance-continue-async-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/external-paintserver-reference-expected.png: Removed.
* LayoutTests/platform/mac/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/external-paintserver-reference-expected.svg: Added.
* LayoutTests/svg/custom/external-paintserver-reference-expected.txt: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.png: Removed.
* LayoutTests/svg/custom/uri-reference-handling-expected.svg: Added.
* LayoutTests/svg/custom/uri-reference-handling-expected.txt: Removed.
* LayoutTests/svg/custom/uri-reference-handling.svg:

Canonical link: https://commits.webkit.org/316417@main
